### PR TITLE
devices: add stm32mp157c-dk2

### PR DESCRIPTION
**Pull request by roxell**
Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

### CODE REVIEW

File: devices/stm32mp157c-dk2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{% set PROJECT = PROJECT|default("") %}
+{% extends PROJECT+"qemu.jinja2" %}
+
+{% set DOWNLOAD_COMMANDS = DOWNLOAD_COMMANDS|default("udhcpc; cd /tmp; wget") %}
+{% set UNPACK_COMMANDS = UNPACK_COMMANDS|default("tar -C / -xzf") %}
+{% set DEPLOY_OS = DEPLOY_OS|default("oe") %}
+{% set BOOT_OS_PROMPT = BOOT_OS_PROMPT|default("root@stm32mp1:~") %}
+{% set DEPLOY_TARGET = DEPLOY_TARGET|default("flasher") %}
+
```
Comment (K1504296) [inline devices/stm32mp157c-dk2:9]: {% set boot_method = boot_method|default("minimal") %}

{% set auto_login = false %}

**Review by K1504296 — CHANGES_REQUESTED** [review]
These additions are required while building the job definition in order to have it deploy to the stm32mp1.

File: devices/stm32mp157c-dk2 [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{% extends PROJECT+"qemu.jinja2" %}
+
+{% set DOWNLOAD_COMMANDS = DOWNLOAD_COMMANDS|default("udhcpc; cd /tmp; wget") %}
+{% set UNPACK_COMMANDS = UNPACK_COMMANDS|default("tar -C / -xzf") %}
+{% set DEPLOY_OS = DEPLOY_OS|default("oe") %}
+{% set BOOT_OS_PROMPT = BOOT_OS_PROMPT|default("root@stm32mp1") %}
+{% set DEPLOY_TARGET = DEPLOY_TARGET|default("flasher") %}
+
+{% set boot = boot|default(false) %}
+{% set use_context = unpack_commands|default(false) %}
```
Comment (mwasilew) [inline devices/stm32mp157c-dk2:11]: I don't understand this. Where is the 'unpack_commands' variable coming from?

**Review by mwasilew — CHANGES_REQUESTED** [review]

File: devices/stm32mp157c-dk2 (hunk 2 above)
Comment (roxell) [reply devices/stm32mp157c-dk2]: I don't understand that either... fixing! =)

**Comment by mwasilew** [comment]
stm32mp1 deriving from qemu doesn't sound right, but I guess this is the closest we can get without creating special 'flasher' template. I'll go with that.

**Pull request merged**